src/Microsoft.AspNet.Server.Kestrel.Https/HttpsConnectionFilter.cs

```diff
@@ -3,29 +3,37 @@
 
 using System;
 using System.Net.Security;
+using System.Security.Authentication;
 using System.Security.Cryptography.X509Certificates;
 using System.Threading.Tasks;
+using Microsoft.AspNet.Http.Features;
+using Microsoft.AspNet.Http.Features.Internal;
 using Microsoft.AspNet.Server.Kestrel.Filter;
 
 namespace Microsoft.AspNet.Server.Kestrel.Https
 {
     public class HttpsConnectionFilter : IConnectionFilter
     {
-        private readonly X509Certificate2 _cert;
+        private readonly X509Certificate2 _serverCert;
+        private readonly ClientCertificateMode _clientCertMode;
+        private readonly ClientCertificateValidationCallback _clientValidationCallback;
         private readonly IConnectionFilter _previous;
+        private X509Certificate2 _clientCert;
 
-        public HttpsConnectionFilter(X509Certificate2 cert, IConnectionFilter previous)
+        public HttpsConnectionFilter(HttpsConnectionFilterOptions options, IConnectionFilter previous)
         {
-            if (cert == null)
+            if (options.ServerCertificate == null)
             {
-                throw new ArgumentNullException(nameof(cert));
+                throw new ArgumentNullException(nameof(options.ServerCertificate));
```

Comment:
This is fine.
Make it an `ArgumentException("The server certificate parameter is required.")`, ServerCertificate is not a direct argument of HttpsConnectionFilter.